### ISSUE TITLE
Stealthed things are visible to the enemy on spawn for a brief moment

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#702]
Stealthed things are visible to the enemy on spawn for a brief moment. For example China mines can be seen after research completion just long enough to know there are mines.

**Comment by MTKing4** [comment]
I think this is intended, not a bug
Leave it as it is

**Comment by xezon** [comment]
Nothing is intentional in this game.

**Comment by ZekeDlyoung** [comment]
That's just how Stealth delay in the stealth module works. You can give them a stealth delay of 0, but then they will instantly disappear once a scan is finished (which I guess is not that much of a problem for mines)

**Comment by xezon** [comment]
Whats the exact name of Stealth Modules?

**Comment by commy2** [comment]
The module is named StealthUpdate.

The issue is, that China mines already have a StealthDelay of 0, yet apparently are still visible to the enemy for a few frames.

This seems to be an engine issue.  Any unit with StealthDelay > 0 on the other hand should be visible while they leave the factory imho.

**Comment by xezon** [comment]
I wonder if a negative `StealthDelay` value would magically change anything.

**Comment by xezon** [comment]
Brooks:
> USA Spy Drone is briefly visible when being placed

**Comment by xezon** [comment]
Maybe this is not a bug if this behavior is consistent for all stealthed objects.

**Comment by Stubbjax** [comment]
> Maybe this is not a bug if this behavior is consistent for all stealthed objects.

The current behaviour is still unintentional and unavoidable. A designer should be able to expect a `StealthDelay` of 0 to mean the object is not visible for a single frame. If a brief window of visibility is desired, then a low value > 0 would do so.

I imagine the Spy Drone being visible for a single frame is even harder to detect with a higher logic tick rate. Such variability is also not ideal.

**Comment by xezon** [comment]
> China mines already have a StealthDelay of 0

Right. That should mean they should be invisible right away. But maybe it is desirable to bump that value after the code fix to reward the attentitiveness when observing the enemy base.

**Comment by xezon** [comment]
We should gather a list of all objects that have StealthDelay = 0
